### PR TITLE
Blast filtering logic

### PR DESCRIPTION
**Pull request by its-mahdi**
- [x] What is the link to the Jira that this PR is for?
https://www.ebi.ac.uk/panda/jira/browse/TRM-24302
- [ ] Have you written tests and do these have >= 75% coverage of the code that you are contributing?
No. Please make suggestions where and what to test.
- [x] Is it possible to visually evaluate this PR in the browser? If so, what are the steps to do this (eg specific URLs or sections)?
Go to a Blast results page. You should see the histograms on the sidebar.
- [ ] Are there any linting errors that required you to add exceptions?
- [ ] If there are backend requests, have you checked for the existence of the resource's attributes before attempting access?

### CODE REVIEW

File: src/tools/blast/components/results/BlastResult.tsx [hunk 1]
```diff
@@ -42,6 +42,11 @@ const BlastResultToolInput = lazy(() =>
     /* webpackChunkName: "blast-result-tool-input" */ './BlastResultToolInput'
   )
 );
+const BlastResultHitDistribution = lazy(() =>
+  import(
+    /* webpackChunkName: "blast-result-tool-input" */ './BlastResultHitDistribution'
```
Comment (aurel-l) [inline src/tools/blast/components/results/BlastResult.tsx:47]:
```suggestion
    /* webpackChunkName: "blast-result-hit-distribution" */ './BlastResultHitDistribution'
```

File: src/tools/blast/components/results/BlastResultsParametersFacets.tsx [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    );
+  };
+
+  const onBlastParamChange = (paramName: string, values: number[]) => {
+    addFacet(paramName, values.join('-'));
+  };
+
+  const binSize = 100;
+
+  const histogramSettings = {
```
Comment (xwatkins) [inline src/tools/blast/components/results/BlastResultsParametersFacets.tsx:91]: These could go up to parent component and be shared with the tab visualisation

**Comment by its-mahdi** [comment]
@aurel-l Do you have any suggestion if code needs to be moved around? I'm not sure if everything there is in the right files. There are also some similar functionality between the `Facets` component and the one that I have implemented. e.g. `addFacet`, etc. Tried to combine these two together, but the Facets component has built with a different goal in mind I guess, so it proved to be difficult.

Let me know if you think any typings should be added or moved into different files too.

**Comment by its-mahdi** [comment]
Another branch ahead of this PR is already merged to 08.

**Pull request closed without merging**